okhttp/src/main/java/okhttp3/HttpUrl.java

```diff
@@ -915,10 +915,16 @@ public Builder newBuilder() {
    * Returns a new {@code HttpUrl} representing {@code url} if it is a well-formed HTTP or HTTPS
    * URL, or throws an exception if it isn't.
    *
+   * @param url string for check URL format
+   * @return instance of {@link HttpUrl}
+   *
    * @throws MalformedURLException if there was a non-host related URL issue
    * @throws UnknownHostException if the host was invalid
+   * @throws IllegalArgumentException if URL not specified
    */
-  static HttpUrl getChecked(String url) throws MalformedURLException, UnknownHostException {
+  public static HttpUrl getChecked(String url)
```

Comment:
Instead of making this public I prefer a new method, `HttpUrl.get(String)` that throws an `IllegalArgumentException` if the input is malformed.

With the right documentation our users should be able to remember that `parse()` returns null and `get()` throws on errors.